### PR TITLE
Turns Deer into Basic Mob - They Freeze At The Sight of Vehicles

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

deers only show up in the BEPIS but i decided that they would be easy enough to turn into a basic mob (they were). it was so easy in fact that i decided to dip my toes into coding AI behavior, and made them freeze up whenever they see a vehicle. this required a lot of code in a bunch of places that i was quite unfamiliar with before starting this project, so do let me know if i glonked up anywhere and i can work on smoothing it out.
## Why It's Good For The Game

one less simple animal on the list. deers staring at headlights is pretty cool i think, neato interaction for when you do get them beyond the joke the bepis makes

i'm also amenable to dropping the whole "deer in headlights" code if you don't like that for w/e reason- just wanted to make them basic at the very least
## Changelog
:cl:
add: If you ever happen upon a wild deer, try not to ride your fancy vehicles too close to it as it'll freeze up like a... you know where I'm going with this.
/:cl:

### CODE REVIEW

**Comment by IHateGeese** [comment]
blends perfectly with the clown car deer PR

File: code/controllers/subsystem/movement/movement_types.dm [hunk 1]
```diff
@@ -739,6 +739,27 @@
...
+ * delay - How many deci-seconds to wait between fires. Defaults to the lowest value, 0.1
+ * timeout - Time in deci-seconds until the moveloop self expires. Defaults to infinity
+ * subsystem - The movement subsystem to use. Defaults to SSmovement. Only one loop can exist for any one subsystem
+ * priority - Defines how different move loops override each other. Lower numbers beat higher numbers, equal defaults to what currently exists. Defaults to MOVEMENT_DEFAULT_PRIORITY
+ * flags - Set of bitflags that effect move loop behavior in some way. Check _DEFINES/movement.dm
+ */
+/datum/controller/subsystem/move_manager/proc/freeze(moving, halted_turf, delay, timeout, subsystem, priority, flags, datum/extra_info)
+	return add_to_loop(moving, subsystem, /datum/move_loop/freeze, priority, flags, extra_info, delay, timeout, halted_turf)
+
+/datum/move_loop/freeze
```
Comment (Mothblocks) [inline code/controllers/subsystem/movement/movement_types.dm:759]: Document

File: code/datums/ai/basic_mobs/basic_ai_behaviors/stop_and_stare.dm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	var/datum/weakref/weak_target = controller.blackboard[target_key]
+	var/atom/movable/target = weak_target?.resolve()
+	if(!ismovable(target) || !isturf(target.loc)) // just to make sure that nothing funky happened between setup and perform
+		return
+
+	var/mob/pawn_mob = controller.pawn
+	var/turf/pawn_turf = get_turf(pawn_mob)
+
+	pawn_mob.face_atom(target)
+	pawn_mob.balloon_alert_to_viewers("stops and stares...")
```
Comment (Mothblocks) [inline code/datums/ai/basic_mobs/basic_ai_behaviors/stop_and_stare.dm:25]: You don't have to do anything now, but in the future what I want us to use is a form of `manual_emote` that performs the Runechat animation, but doesn't put it in the chat.

File: code/datums/ai/basic_mobs/basic_subtrees/stare_at_movable.dm [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/// Locate a movable to stop and stare at.
+/datum/ai_planning_subtree/stare_at_movable
+
+/datum/ai_planning_subtree/stare_at_movable/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/datum/weakref/weak_target = controller.blackboard[BB_STATIONARY_CAUSE]
+	var/atom/target = weak_target?.resolve()
+
+	if(!ismovable(target))
+		var/list/potential_scares = controller.blackboard[BB_STATIONARY_TARGETS]
+		controller.queue_behavior(/datum/ai_behavior/find_and_set/in_list, BB_STATIONARY_CAUSE, potential_scares)
```
Comment (Mothblocks) [inline code/datums/ai/basic_mobs/basic_subtrees/stare_at_movable.dm:10]: What is this codepath doing, and why do we have cases where we're not setting this to a movable?

File: code/modules/mob/living/basic/farm_animals/deer.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	var/time_to_freeze_for = (rand(5, 10) SECONDS)
+	ai_controller.blackboard[BB_STATIONARY_SECONDS] = time_to_freeze_for
+	ai_controller.blackboard[BB_STATIONARY_COOLDOWN] = (time_to_freeze_for * (rand(3, 5)))
+	ai_controller.blackboard[BB_STATIONARY_TARGETS] = stationary_scary_things
+
+/datum/ai_controller/basic_controller/deer
+	blackboard = list(
+		BB_BASIC_MOB_FLEEING = TRUE,
+		BB_STATIONARY_MOVE_TO_TARGET = TRUE,
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/ignore_faction(),
```
Comment (Mothblocks) [inline code/modules/mob/living/basic/farm_animals/deer.dm:39]:
```suggestion
		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/ignore_faction,
```

File: code/modules/mob/living/basic/farm_animals/deer.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/mob/living/basic/deer
+	name = "doe"
+	desc = "A gentle, peaceful forest animal. How did this get into space?"
+	icon_state = "deer-doe"
+	icon_living = "deer-doe"
+	icon_dead = "deer-doe-dead"
+	gender = FEMALE
+	mob_biotypes = MOB_ORGANIC|MOB_BEAST
+	speak_emote = list("grunts","grunts lowly")
```
Comment (Mothblocks) [inline code/modules/mob/living/basic/farm_animals/deer.dm:9]:
```suggestion
	speak_emote = list("grunts", "grunts lowly")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**